### ISSUE TITLE
twist on python3.5 fails when given no plugin

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8918]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)| @moshez reported|
|-|-|
|Trac ID|trac#8918|
|Type|defect|
|Created|2016-11-25 17:42:06Z|
|Branch|https://github.com/twisted/twisted/tree/8918-twist-no-plugin|

Git hash: 7314e23b638ae2f59cce91107c41bd4567789044

Python 3.5: Ubuntu, python3.5 (3.5.0-3)

```
Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'win32er'

Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gi'

Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'iocp'

Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk2'

Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk3'

Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'wx'

Unhandled Error
Traceback (most recent call last):
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
    
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 28, in <module>
    class TwistOptions(Options):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in TwistOptions
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 66, in <genexpr>
    '"{}"'.format(rt.shortName) for rt in getReactorTypes()
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'glib2'

Unhandled Error
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 157, in parseOptions
    self.installReactor()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'win32er'

Unhandled Error
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 157, in parseOptions
    self.installReactor()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gi'

Unhandled Error
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 157, in parseOptions
    self.installReactor()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'iocp'

Unhandled Error
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 157, in parseOptions
    self.installReactor()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk2'

Unhandled Error
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 157, in parseOptions
    self.installReactor()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'gtk3'

Unhandled Error
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 157, in parseOptions
    self.installReactor()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 77, in installReactor
    self["reactor"] = installReactor(name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/reactors.py", line 80, in installReactor
    for installer in getReactorTypes():
--- <exception caught here> ---
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 217, in getPlugins
    adapted = interface(plugin, None)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 131, in __call__
    adapter = self._call_conform(conform)
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/lib/python3.5/site-packages/zope/interface/interface.py", line 467, in _call_conform
    return conform(self)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 71, in __conform__
    return self.load()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/plugin.py", line 66, in load
    return namedAny(self.dropin.moduleName + '.' + self.name)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/reflect.py", line 311, in namedAny
    obj = getattr(obj, n)
builtins.AttributeError: module 'twisted.plugins.twisted_reactors' has no attribute 'wx'

Error: No plugin specified.

Usage: twist [options] plugin [plugin_options]
Options:
      --log-file=    Log to file. ("-" for stdout, "+" for stderr; default: "-")
      --log-format=  Log file format. (options: "text", "json"; default: "text"
                     if the log file is a tty, otherwise "json")
      --version      Print version and exit.
      --help         Display this help and exit.
      --log-level=   Set default log level. (options: "debug", "info", "warn",
                     "error", "critical"; default: "info")
      --reactor=     The name of the reactor to use. (options: "kqueue", "poll",
                     "select", "epoll", "default", "cf")
Commands:
    conch            A Conch SSH service.
    dns              A domain name server.
    ftp              An FTP server.
    inetd            An inetd(8) replacement.
    mail             An email service
    manhole          An interactive remote debugger service accessible via
                     telnet and ssh and providing syntax coloring and basic line
                     editing functionality.
    news             A news server.
    portforward      A simple port-forwarder.
    procmon          A process watchdog / supervisor
    socks            A SOCKSv4 proxy service.
    web              A general-purpose web server which can serve from a
                     filesystem or application resource.
    words            A modern words server
    xmpp-router      An XMPP Router server
Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 38, in options
    options.parseOptions(argv[1:])
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 160, in parseOptions
    Options.parseOptions(self, options=options)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/python/usage.py", line 277, in parseOptions
    self.postOptions()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_options.py", line 194, in postOptions
    raise UsageError("No plugin specified.")
twisted.python.usage.UsageError: No plugin specified.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/moshez/src/twisted/github/twisted/build/py35-venv/bin/twist", line 9, in <module>
    load_entry_point('Twisted==16.5.0', 'console_scripts', 'twist')()
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 132, in main
    options = cls.options(argv)
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/twist/_twist.py", line 40, in options
    exit(ExitStatus.EX_USAGE, "Error: {}\n\n{}".format(e, options))
  File "/home/moshez/src/twisted/github/twisted/src/twisted/application/runner/_exit.py", line 36, in exit
    out.write(b"\n")
TypeError: write() argument must be str, not bytes
```

Expected output:

```
(py27-venv)moshez@mcgyver:~/src/twisted/github/twisted$ twist
Error: No plugin specified.

Usage: twist [options] plugin [plugin_options]
Options:
      --log-level=   Set default log level. (options: "debug", "info", "warn",
                     "error", "critical"; default: "info")
      --help         Display this help and exit.
      --version      Print version and exit.
      --log-format=  Log file format. (options: "text", "json"; default: "text"
                     if the log file is a tty, otherwise "json")
      --log-file=    Log to file. ("-" for stdout, "+" for stderr; default: "-")
      --reactor=     The name of the reactor to use. (options: "win32", "gi",
                     "kqueue", "poll", "iocp", "select", "epoll", "gtk2",
                     "gtk3", "wx", "default", "cf", "glib2")
Commands:
    conch            A Conch SSH service.
    dns              A domain name server.
    ftp              An FTP server.
    inetd            An inetd(8) replacement.
    mail             An email service
    manhole          An interactive remote debugger service accessible via
                     telnet and ssh and providing syntax coloring and basic line
                     editing functionality.
    news             A news server.
    portforward      A simple port-forwarder.
    procmon          A process watchdog / supervisor
    socks            A SOCKSv4 proxy service.
    web              A general-purpose web server which can serve from a
                     filesystem or application resource.
    words            A modern words server
    xmpp-router      An XMPP Router server


```

<details><summary>Searchable metadata</summary>

```
trac-id__8918 8918
type__defect defect
reporter__moshez moshez
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8918_twist_no_plugin 8918-twist-no-plugin
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1480095726800754 1480095726800754
changetime__1480383795652175 1480383795652175
version__None None
owner__moshez moshez

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

The stuff above twist is because you're running from a checkout (so it's expected).

PR for the bottom issue (the bytes/unicode thing) is at https://github.com/twisted/twisted/pull/605 .

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_3"></a>|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

reviewed, good to merge

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @moshez<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

In changeset 889a4db888c9f4290dd2e4e40f6905c70fe43b86
```
#!CommitTicketReference repository="" revision="889a4db888c9f4290dd2e4e40f6905c70fe43b86"
Merge 8918-twist-no-plugin: Fix twist on Python 3 when no plugin is chosen

Author: hawkowl
Reviewer: wsanchez
Fixes: #8918
```
